### DISCUSSION TITLE
Can Micronaut's reflection-free Jackson module be used independently?

### DISCUSSION THREAD

**Discussion by kaqqao** [micronaut-projects/micronaut-core#6486] [Q&A]
I was looking into reflection-less Jackson, and discovered that Micronaut already has a great solution. But I'm not sure how friendly it is towards usage independent of Micronaut itself.
Is it doable at all? Is there perhaps a test or something like that I can look at to get an idea?

Edit: I see that it depends on micronaut-context and micronaut-http which are pretty heavy dependencies... Which is a bummer 😓

**Comment by yawkat** [comment]
You could get away without http and maybe without context. http is only used for some property name special handling and could be removed, and I don't think context is needed by the `BeanIntrospectionModule` at all (though it is required by the rest of the `micronaut-jackson-databind` module).

You will not get away without `micronaut-core` though, since introspection support is obviously required for `BeanIntrospectionModule`.
